rapids-cmake/cpm/detail/get_proprietary_binary_url.cmake

```diff
@@ -61,25 +60,28 @@ function(rapids_cpm_get_proprietary_binary package_name version)
     include("${rapids-cmake-dir}/rapids-version.cmake")
   endif()
 
+  # Determine the CUDA Toolkit version so that we properly evaluate the placeholders in
+  # `proprietary_binary`
+  if(proprietary_binary MATCHES "{cuda-toolkit-version")
+    if(CUDAToolkit_VERSION_MAJOR)
+      set(cuda-toolkit-version ${CUDAToolkit_VERSION_MAJOR}.${CUDAToolkit_VERSION_MINOR})
+      set(cuda-toolkit-version-major ${CUDAToolkit_VERSION_MAJOR})
+    elseif(CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA" AND CMAKE_CUDA_COMPILER_VERSION MATCHES
+                                                        [=[([0-9]+)\.([0-9]+)\.([0-9]+)]=])
+      # Use if(MATCHES) so we get the MATCH_`N` variables. Use same regex as used by FindCUDAToolkit
+      set(cuda-toolkit-version-major "${CMAKE_MATCH_1}")
+      set(cuda-toolkit-version "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}")
+    else()
```

Comment:
When does the else branch get hit? The first branch is if the calling code has already found CUDA, but the second and third seem to be redundant ways of handling other instances where CUDA code is being compiled. Why do we need both, and is this particular sequence important?

Reply:
2nd is when the CUDA language hasn't been enabled but  `find_package(CUDAToolkit)` has already been called. The else would only happen when neither of the two is true.

But yes we can reduce this by removing the `elseif` entirely and just have the possibly uneeded `find_package(CUDAToolkit)` call.

Reply:
Removed the old comment.

The second entry hits when we have enabled the `CUDA` language and not found the CUDA toolkit.
But yes I was being fancy and we should just call `find_package(CUDAToolkit)`